### PR TITLE
fix: [CDS-81020]:  Use ModalDialog inplace of Dialog for NewEditServiceModal

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary
**Issue:** Service Modal data reset if there is an error in any of the API.

**Reason:** We were using PageSpinner to show the loader for the API call and form mount again with the initial values so the field values get empty.

**Fix:** Use updated ModalDialog in place of Dialog for the NewEditServiceModal

#### Screenshots
**Before:**

https://github.com/harness/harness-core-ui/assets/110591046/fa95a001-7b4c-4dec-a01f-e09202540d64

**After:**

https://github.com/harness/harness-core-ui/assets/110591046/a7baaeef-7a1f-42dd-94be-261a1b03c6ee

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

**Comment by pankajsingh-harness** [comment]
trigger messagecheck

File: src/modules/75-cd/components/PipelineSteps/DeployServiceStep/__tests__/DeployServiceStep.test.tsx [hunk 1]
```diff
@@ -367,4 +368,35 @@ describe('ServiceModal ', () => {
...
+    )
+
+    const $input = getByPlaceholderText('common.namePlaceholder')
+    const $save = getByText(container, 'save')
+
+    fireEvent.change($input, { target: { value: 'test-service' } })
+    fireEvent.click($save)
+
+    const spinner = container.querySelector('.bp3-spinner')
+    expect(spinner).toBeTruthy()
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/DeployServiceStep/__tests__/DeployServiceStep.test.tsx:400]: Can we use toBeInTheDocument here ?

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/DeployServiceStep/__tests__/DeployServiceStep.test.tsx (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceStep/__tests__/DeployServiceStep.test.tsx]: Done.

**Review by chetanvekariya-harness — DISMISSED** [review]

**Comment by sunnykesh-harness** [comment]
We should cover entire modal with spinner like pipeline create form as discuused.

**Review by rameshx — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Review by srdjan-harness — APPROVED** [review]

**Pull request merged**